compiler/basicTypes/Var.hs-boot

```diff
@@ -0,0 +1,12 @@
+module Var where
+
+import GhcPrelude
+import Outputable
+import Data.Data
+
+data Var
+type Id = Var
+type TyVar = Id
+type TyCoVar = Id
+
+data ArgFlag
```

Comment:
Curious. Why is it needed now, but wasn't before?

Reply:
After `Multiplicity` imports `TysWiredIn`, we're getting an import cycle:

```
         module ‘Var’ (compiler/basicTypes/Var.hs)
        imports ‘Multiplicity’ (compiler/basicTypes/Multiplicity.hs)
  which imports ‘TysWiredIn’ (compiler/prelude/TysWiredIn.hs-boot)
  which imports ‘Var’ (compiler/basicTypes/Var.hs)
```

Ideally, we would create a `hs-boot` file for `Multiplicity`. However, `Var` is using `Omega`, which means `Omega` would have to be in the `hs-boot` file, and currently pattern synonyms are not supported in hs-boot files. Since we can't change Var -> Multiplicity import, and Multiplicity -> TysWiredIn is already a boot import, this means TysWiredIn has to import `Var.hs-boot`

This `Var.hs-boot` existed in an earlier version of the branch, I removed it in 272d7605530 because it was not used.